src/components/character-controller.js

```diff
@@ -114,6 +114,9 @@ AFRAME.registerComponent("character-controller", {
       pivotRotationMatrix.makeRotationAxis(rotationAxis, pivot.rotation.y);
       pivotRotationInvMatrix.makeRotationAxis(rotationAxis, -pivot.rotation.y);
       this.updateVelocity(deltaSeconds);
+      if (this.velocity.x === 0 && this.velocity.y === 0 && this.velocity.z === 0) {
```

Comment:
I would use the EPS constant (https://en.wikipedia.org/wiki/Machine_epsilon)
Actually since it's already being used down below, I would just cache if this.velocity.lengthSq() and use it in both places
```
const magnitude = this.velocity.lengthSq();
if (magnitude < EPS)